cocos/animation/marionette/animation-graph-context.ts

```diff
@@ -106,11 +129,46 @@ export class AnimationGraphBindingContext {
         return this._varRegistry[id];
     }
 
+    /**
+     * Pushes the `additive` flag. A later `_popAdditiveFlag` is required to pop the change.
+     * @internal
+     */
+    public _pushAdditiveFlag (additive: boolean) {
+        this._additiveFlagStack.push(additive);
+    }
+
+    /**
+     * Undo last `_pushAdditiveFlag`.
+     * @internal
+     */
+    public _popAdditiveFlag () {
+        assertIsTrue(this._additiveFlagStack.length > 1);
+        this._additiveFlagStack.pop();
+    }
+
+    /** @internal */
+    public _integrityCheck () {
+        return this._additiveFlagStack.length === 1;
+    }
+
     private _origin: Node;
 
     private _layoutMaintainer: AnimationGraphPoseLayoutMaintainer;
 
-    private _varRegistry: VarRegistry
+    private _varRegistry: VarRegistry;
```

Comment:
Should use `declare`?

Reply:
Why does it?

Reply:
Does it affect performance if not having "declare"?